apps/render_cf.py

```diff
@@ -234,6 +234,11 @@ def validate_job_spec(job_type: str, job_spec: dict) -> None:
         raise ValueError(f"{job_type} contains reserved parameter name 'job_id'")
 
     expected_param_fields = ['api_schema']
+
+    for step in job_spec['steps']:
+        if not step['image'].islower():
+            raise ValueError(f'{job_type} has image {step["image"]} and docker requires the image to be all lowercase')
```

Comment:
Let's move this check to the end of the function, since I think it makes more sense to validate the steps *after* validating the job spec parameters.